operator/v1/types_machineconfiguration.go

```diff
@@ -153,3 +167,181 @@ const (
+	// +optional
+	// +listType=map
+	// +listMapKey=path
+	Files []NodeDisruptionPolicyFile `json:"files,omitempty"`
+	// units is a list MachineConfig unit definitions and actions to take on changes to those services
+	// +optional
+	// +listType=map
+	// +listMapKey=name
+	Units []NodeDisruptionPolicyUnit `json:"units,omitempty"`
+	// sshkey is the overall sshkey MachineConfig definition
+	// +optional
+	SSHKey NodeDisruptionPolicySSHKey `json:"sshkey,omitempty"`
+}
+
+// NodeDisruptionPolicyFile is a file entry and corresponding actions to take
+type NodeDisruptionPolicyFile struct {
+	// path represents the filepath
+	// +kubebuilder:validation:Required
+	Path string `json:"path"`
+	// actions represents the series of commands to be executed on changes to the corresponding type and value
+	// +kubebuilder:validation:Required
+	// +listType=atomic
+	Actions []NodeDisruptionPolicyAction `json:"actions"`
+}
+
+// NodeDisruptionPolicyUnit is a systemd unit name and corresponding actions to take
+type NodeDisruptionPolicyUnit struct {
+	// name represents the service name
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+	// actions represents the series of commands to be executed on changes to the corresponding type and value
+	// +kubebuilder:validation:Required
+	// +listType=atomic
+	Actions []NodeDisruptionPolicyAction `json:"actions"`
+}
+
+// NodeDisruptionPolicySSHKey is actions to take for any SSHKey change
+type NodeDisruptionPolicySSHKey struct {
+	// actions represents the series of commands to be executed on changes to the corresponding type and value
+	// +kubebuilder:validation:Required
+	// +listType=atomic
+	Actions []NodeDisruptionPolicyAction `json:"actions"`
+}
+
+// Validation is a string enum used in a NodeDisruptionPolicyStatus object. They describe whether the controller
+// was able to validate the latest spec
+// +kubebuilder:validation:Enum:="success";"failed"
+type Validation string
+
+const (
+	// success represents a valid cluster node disruption policy
+	Success Validation = "success"
+
+	// failed represents an invalid cluster node disruption policy
+	Failed Validation = "failed"
+)
+
+// NodeDisruptionPolicyType is a string enum used in a NodeDisruptionPolicyConfig object. They describe the type of
+// the object that this policy will be in effect for.
+// +kubebuilder:validation:Enum:="file";"service";"sshKey"
+type NodeDisruptionPolicyType string
+
+const (
+	// File represents a NodeDisruption policy that is in effect for changes to a file.
+	File NodeDisruptionPolicyType = "file"
+
+	// Service represents a NodeDisruption policy that is in effect for changes to a service.
+	Service NodeDisruptionPolicyType = "service"
+
+	// Service represents a NodeDisruption policy that is in effect for changes to a sshKey
+	SSHKey NodeDisruptionPolicyType = "sshKey"
+)
+
+// TODO maybe do this in the MCC validation and skip here?
```

Comment:
Prefer to keep as much API validation at admission time as possible rather than later, either this should be done here or in a webhook, CEL is preferable in this case IMO